framework/decode/metadata_consumer_base.h

```diff
@@ -103,6 +106,11 @@ class MetadataConsumerBase
     virtual void ProcessInitSubresourceCommand(const format::InitSubresourceCommandHeader& command_header,
                                                const uint8_t*                              data)
     {}
+
+    inline void SetCurrentBlockIndex(uint64_t block_index)
```

Comment:
Declaring this in the common (between dx12 and Vulkan) consumer class is a good idea.
Since `Dx12ConsumerBase` inherits from `MetadataConsumerBase` needs to be also updated:
- `Dx12ConsumerBase::SetCurrentBlockIndex` needs to be marked `vitual` + `override`
- `Dx12ConsumerBase::current_block_index_` needs to be removed as it will inherit the newly added `block_index_` from this class